rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -57,16 +57,16 @@ Suppose a CBMT with *n* items, the size of the array would be *2n-1*, the index
 
 ## Merkle Proof
 
-Merkle Proof can provide a proof for existence of one or more items. Only sibling of the nodes along the path that form leaves to root, excluding the nodes already in the path, should be included in the proof. We also specify that ***the nodes in the proof is presented in descending order***(with this, algorithms of proof's generation and verification could be much simple). For example, if we want to show that `[T1, T4]` is in the list of 6 items above, only nodes `[T5, T0, B3]` should be included in the proof.
+Merkle Proof can provide a proof for existence of one or more items. Only sibling of the nodes along the path that form leaves to root, excluding the nodes already in the path, should be included in the proof. We also specify that ***the nodes in the proof is presented in descending order***(with this, algorithms of proof's generation and verification could be much simple). The indexes of items that need to prove is also included in the proof, and ***presented in ascending order by corresponding hash***. For example, if we want to show that `[T1, T4]` is in the list of 6 items above, only nodes `[T5, T0, B3]` and indexes `[9, 6]` should be included in the proof.
```

Comment:
Why is it more reasonable to include item indices in proof?

Reply:
Because the tree index will be transparent to the caller, and the caller doesn't need to care about the tree index.

Reply:
Can you be more concrete and give an example? I need to understand the problem and context first to understand the change.

Reply:
In the former design, the caller needs to know the index of each item, but in some scenarios, the caller may have no idea about the index, so the caller may need extra interaction to query the index. It will make the proof verification complicate and interactive.  Ideally, the caller should verify the items without any extra information, beyond the items he has and proof he got.

Reply:
What's the scenario you refer to? Is it worth to add the scenario description and the reason of including index in proof into rfc?

Reply:
In the scenario that index is not known, for example, in most cases, the SPV client got a transaction, since the index is not the feature of transactions, the client has no idea about the index.  So the client has to query the index. But the index is the feature of the Merkle tree, it should be transparent to the client.